src/Microsoft.Health.Fhir.Core/Features/Operations/Reindex/ReindexJobTask.cs

```diff
@@ -105,21 +109,36 @@ public async Task ExecuteAsync(ReindexJobRecord reindexJobRecord, WeakETag weakE
                     var resourceList = new HashSet<string>();
                     foreach (var param in notYetIndexedParams)
                     {
-                        if (param.TargetResourceTypes != null)
+                        foreach (var baseResourceType in param.BaseResourceTypes)
                         {
-                            resourceList.UnionWith(param.TargetResourceTypes);
-                        }
+                            if (baseResourceType == "Resource")
+                            {
+                                resourceList.UnionWith(_modelInfoProvider.GetResourceTypeNames().ToHashSet());
 
-                        if (param.BaseResourceTypes != null)
-                        {
-                            resourceList.UnionWith(param.BaseResourceTypes);
+                                // We added all possible resource types, so no need to continue
+                                break;
+                            }
+
+                            if (baseResourceType == "DomainResource")
+                            {
+                                resourceList.UnionWith(_modelInfoProvider.GetResourceTypeNames().ToHashSet());
+
+                                // Remove types that don't inherit from domain resource
+                                resourceList.Remove("Binary");
+                                resourceList.Remove("Bundle");
+                                resourceList.Remove("Parameters");
+                            }
+                            else
+                            {
+                                resourceList.UnionWith(new HashSet<string> { baseResourceType });
```

Comment:
>new HashSet<string>  [](start = 55, length = 20)

I would go with `new[]{baseResourceType}` or maybe ``Enumerable.Repeat(baseResourceType,1)`;

Reply:
Fixed in 7555909.